app/domain/usecase/get_deliveries.go

```diff
@@ -0,0 +1,44 @@
+package usecase
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/techhub-jf/farmacia-back/app/domain/dto"
+	"github.com/techhub-jf/farmacia-back/app/gateway/api/handler/schema"
+)
+
+type GetDeliveriesInput struct {
+	Pagination dto.Pagination
+}
+
+type GetDeliveriesOutput struct {
+	schema.ListDeliveriesOutput
+}
+
+func (u *UseCase) GetDeliveries(ctx context.Context, input GetDeliveriesInput) (GetDeliveriesOutput, error) {
+	args := dto.Pagination{
+		Page:         input.Pagination.Page,
+		SortBy:       input.Pagination.SortBy,
+		SortType:     input.Pagination.SortType,
+		ItemsPerPage: input.Pagination.ItemsPerPage,
+	}
+
+	deliveries, totalRecords, err := u.DeliveriesRepository.ListAll(ctx, args)
```

Comment:
```suggestion
	deliveries, totalRecords, err := u.DeliveriesRepository.ListAll(ctx, input.Pagination)
```

Já temos um instância de pagination no input